File: examples/DesignPatterns/FactoryMethod/typescript/src/models/ActionResult.ts
import { Action } from './Action';
import { ActionType } from './ActionType';

export class ActionResult<T> extends Action<T> {
  private _dispatcherID: number;

  constructor(type: ActionType, payload: T, dispatcherID: number) {
    super(type, payload);
    this._dispatcherID = dispatcherID;
  }

  public get dispatcherID(): number {
    return this._dispatcherID;
  }
  public set dispatcherID(value: number) {
    this._dispatcherID = value;
  }
}

File: examples/DesignPatterns/FactoryMethod/src/main/java/bead/ex/FactoryMethod/models/ActionResult.java
package bead.ex.FactoryMethod.models;

import bead.ex.FactoryMethod.enums.ActionType;

public class ActionResult<T> extends Action<T> {
    private int dispatcherID;
    public ActionResult(ActionType type, T payload, int dispatcherID) {
        super(type, payload);
        this.dispatcherID = dispatcherID;
    }

    public int getDispatcherID() {
        return dispatcherID;
    }

    public void setDispatcherID(int dispatcherID) {
        this.dispatcherID = dispatcherID;
    }
}
